connect-inject/container_init.go

```diff
@@ -47,6 +49,17 @@ func (h *Handler) containerInit(pod *corev1.Pod) (corev1.Container, error) {
 		}
 	}
 
+	// If tags are specified split the string into an array and create
+	// the tags string
+	if raw, ok := pod.Annotations[annotationTags]; ok && raw != "" {
+		tags := strings.Split(raw, ",")
+
+		// Create json array from the annotations
+		jsonTags, _ := json.Marshal(tags)
```

Comment:
It's a JSON array but then it's being templated into HCL. Are there any json marshaling vs. HCL issues we might run  into here?

Reply:
I don't think you should swallow the error here. Would be useful to surface instead I think.

Reply:
I've added a log message for this, but I don't think a bad annotation should stop connect injection. This will at least make it possible for folks to see what's happening if it's not what they expect in this situation.

Reply:
From the [hcl readme](https://github.com/hashicorp/hcl): "Arrays can be made by wrapping it in []. Example: ["foo", "bar", 42]", so we should be good for formatting.